### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/advanced/replace.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#786] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/advanced/replace.html

当在 中设置 { custom: true } 时，主题将通过 @theme-hope 别名来引入组件，所以你可以利用这一点来替换主题的任何一个组件。 如何通过别名替换组件 你需要在自己的 VuePress 配置文件通过 alias 替换主题中使用的组件别名。 可以支持的别名如下。 主题组件别名 组件: @theme-hope/components/A...

https://theme-hope.vuejs.press/zh/guide/advanced/replace.html

**Comment by wenjack12** [comment]
config.ts引入：import { getDirname, path } from "vuepress/utils" 后
报错：
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './utils' is not defined by "exports" in G:\project\edu-help\node_modules\vuepress\package.json imported from G:\project\edu-help\src\.vuepress\config.ts.6ec89df5.mjs

**Reply by Mister-Hope** [reply]
先升级到最新版本，再说别的

**Comment by luxiag** [comment]
作者大大，vuepress-theme-hope/lib/shared/index.js 里面的枚举使用替换主题组件要怎么才能获取的到
```ts
import { ArticleInfo, PageInfo as PageInfoEnum, PageType, } from "../../../../shared/index.js";

```

**Reply by Mister-Hope** [reply]
直接导入vuepress-theme-hope/client能获取到吧

**Reply by Mister-Hope** [reply]
哦不对，你直接使用常量就好了
